### DISCUSSION TITLE
Auto accept edits and commands

### DISCUSSION THREAD

**Discussion by asteroideB4612** [cline/cline#635] [Feature Requests]

**Comment by michaelkuhlman** [comment]
For now I suppose you could set a macro on your PC to click the "Approve" button every ~30 seconds? But these features would be awesome to be able to enable/disable.

**Comment by seoeaa** [comment]
this already exists 
![2024-11-01_05-40-32](https://github.com/user-attachments/assets/2c3d0298-8aaf-4827-92ab-6d9a779f1724)

**Reply by KJ7LNW** [reply]
would like to extend this to other operations as well. in fact, I would like to mark some commands as safe. for example, I would rather than it not ask me to run things like `make test` or pytest: 

```
Cline wants to execute this command:
python -m pytest tests/test_openai.py tests/test_google_ai.py tests/test_vertex_ai.py tests/test_anthropic_ai.py -v
```

I would love a check box that says something like `[x] always allow this command to execute without permission`

**Reply by israelmirsky** [reply]
I want this so much right now. Tried to write a macro to click the save button and just keep getting errors.

**Reply by KJ7LNW** [reply]
Even better, have two regex allowlists:
 1. allowlist of commands
    - The `[x] always allow this command to execute without permission` option would add to this list
 2. allowlist of files that can be written without asking
    - Similarly, the `[x] always writing to this file without permission` option would add to this list

Both settings need to be per project, I often have multiple vscode projects open simultaneously.

**Reply by upsidecode** [reply]
> this already exists ![2024-11-01_05-40-32](https://private-user-images.githubusercontent.com/3609621/382167400-2c3d0298-8aaf-4827-92ab-6d9a779f1724.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0MDg0NDQsIm5iZiI6MTczNDQwODE0NCwicGF0aCI6Ii8zNjA5NjIxLzM4MjE2NzQwMC0yYzNkMDI5OC04YWFmLTQ4MjctOTJhYi02ZDlhNzc5ZjE3MjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTdUMDQwMjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTJjYzA4NmM4ZGQ1NjljMTAwNDMwYWEwMjJiY2ZiZTdkMWY5ZDdiZTYzYmUyNjhkYmUxNTE1ZTE0OWZiOGZkNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.uxcNRSdxf489H1BjquH2RtMPfxIm8h4EAgyQmmyHQK8)

where is this setting located?

Update: Ok i found it, on the extension itself

**Comment by stevejford** [comment]
I would like to see it do what bolt.new have implemented. They have files
that that the user can lock and they also have targeted files. That target
the most important files.

**Comment by lloydchang** [comment]
Related https://github.com/cline/cline/discussions/368 https://github.com/cline/cline/discussions/635 https://github.com/cline/cline/pull/674 https://github.com/cline/cline/pull/784 https://github.com/cline/cline/pull/802

**Comment by lloydchang** [comment]
if anyone needs Auto Approve Mode, it's available via Roo Cline at https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

**Comment by saoudrizwan** [comment]
Cline's most requested feature is here 🎉 You can now let Cline work more autonomously with the new Auto-approve menu, letting you choose what tools he can use without needing your permission.

![image](https://github.com/user-attachments/assets/c2cc2e06-abd6-4835-8bdf-0b05283ed7c0)

- Enable or disable for: reading files & directories, editing files, executing terminal commands, using the browser, and using MCP servers.
- Set a limit for how many API requests Cline makes before asking for your approval to keep going.
- And if you have Cline working in the background, he can send system notifications for when he needs your attention, i.e. to get your approval, answer a question, or when a task is completed.

![image](https://github.com/user-attachments/assets/2195f04a-ae37-4dea-a81d-cf9dac5f92a6)
